projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -321,6 +321,10 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         const dataLoaded = (!this._data || this._data.length === 0) && value && value.length > 0;
         this._data = value || [];
         this.summaryService.clearSummaryCache();
+        if (!this._init) {
+            this.validation.updateAll(this._data);
```

Comment:
One slight issue with this - it's missing the "w/ any possible transactions active on top" from my original comment. As in, if there's an active edit on a row in transactions and those are still active, they won't be reflected in this data.
Ideally, I was hoping we can hook into some place along the pipeline, but I don't see a cached result from the transactions pipe like some others do and no easy way to grab the merged data result :/
This will certainly complicate the fix a fair amount, will let you know if I see something useful

Reply:
The updateAll method does get the aggregated values from the transaction service, which should include all transactions (pending and active) :
const recAggregatedData = this.grid.transactions.getAggregatedValue(rowId, true) || rec;
Shouldn't that be enough?